mettle/worker/worker.go

```diff
@@ -199,12 +212,13 @@ func runForever(instanceName, requestQueue, responseQueue, name, storage, dir, c
 		// possible for the subprocesses.
 		runtime.GC()
 		w.Report(true, false, true, "Awaiting next task...")
-		if _, err := w.RunTask(ctx); err != nil {
-			if ctx.Err() != nil {
-				// Error came from a signal triggered above. Give a brief period to send reports then die.
-				time.Sleep(500 * time.Millisecond)
-				return fmt.Errorf("terminated by signal")
-			}
+		_, err := w.RunTask(ctx)
+		if ctx.Err() != nil {
+			// Error came from a signal triggered above. Give a brief period to send reports then die.
+			time.Sleep(500 * time.Millisecond)
```

Comment:
Would be cool if we can unacknowledge the message when this context is cancelled so if it's been shutdown and not completed it'll be republished straight away. Might be a little tricky here though as this function has not knowledge of pubsub messages

Reply:
I agree, will have a think about this